brikkit.js

```diff
@@ -135,84 +140,103 @@ class Brikkit {
             'Plate',
             'Peaks'].indexOf(mapName) === -1)
             return;
-        
+
         this._brickadia.write(`travel ${mapName}\n`);
     }
-    
+
     getScraper() {
         return this._scraper;
     }
-    
+
     getPluginSystem() {
         return this._pluginSystem;
     }
-    
+
     // adds callbacks to print out stdout and stderr directly from Brickadia
     _developmentMode() {
         this._brickadia.on('out', line => console.log(`bout: "${line}"`));
         this._brickadia.on('err', line => console.log(`berr: "${line}"`));
     }
-    
+
     _handleBrickadiaLine(line) {
-        line = stripAnsi(line);
-        
         const matches = /^\[(.*?)\]\[.*?\](.*?): (.*)$/.exec(line);
-        
+
         if(matches === undefined || matches === null)
             return;
-        
+
         const dateString = matches[1]
             .replace(':', '.')
             .replace('-', 'T')
             .replace('.', '-')
             .replace('.', '-')
             .replace('.', ':')
             .replace('.', ':');
-    
+
         const date = new Date(dateString + 'Z');
-        
+
         // which object generated the message
         // UE4 specific: LogConfig, LogInit, ...
         // useful for understanding the line
         const generator = matches[2];
-        
+
         const restOfLine = matches[3];
-        
+
         const joinedPlayer = this._joinParser.parse(generator, restOfLine);
         if(joinedPlayer !== null) {
+            joinedPlayer._brikkit = this;
             this._addPlayer(joinedPlayer);
-            this._putEvent(new Event.JoinEvent(date, joinedPlayer));
+
+            joinedPlayer.getController()
+                .then(() => this._putEvent(new Event.JoinEvent(date, joinedPlayer)))
+                .catch(e => {
+                    this._putEvent(new Event.JoinEvent(date, joinedPlayer))
+                    console.error(e);
+                });
```

Comment:
We could use `this._putEvent(new Event.JoinEvent(date, joinedPlayer)` in `.finally` to not repeat it?